### PR TITLE
create download button to download metadata in blocks

### LINKED ISSUE #3222: Block Viewer: Allow downloading `meta.json` of a block
As all the metadata is already available on the Block Viewer for a block, we can also provide a button to download the `meta.json` for a block.

You can get more idea about programmatic file downloads using JS in [this blog post](https://blog.logrocket.com/programmatic-file-downloads-in-the-browser-9a5186298d5c/).

### PR DESCRIPTION
**Pull request by Oghenebrume50**
Fixes #3222

* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes
- I created a hyperlink tag in the `Blockdetails` component to act as a download button
- I created a helper that returns the URL from the block metadata
 
## Verification
The below screenshots show how it works 
See download button and click on it
download1
Receive the prompt to download file
dowload2
view file after downloading
metapic

### CODE REVIEW

File: CHANGELOG.md
Comment (OGKevin) [inline CHANGELOG.md:915]: Could you be a little more descriptive for the end user on what this PR adds? You can take inspiration from the PR title.

Comment (OGKevin) [reply CHANGELOG.md:915]: Also this needs to be added at the top of the file under the "Unreleased" block.

**Review by OGKevin — CHANGES_REQUESTED** [review]
Minor changes to the CHANGELOG.MD

File: CHANGELOG.md
Comment (Oghenebrume50) [reply CHANGELOG.md]: Thank you, I made the corrections

Comment (OGKevin) [reply CHANGELOG.md]: LGTM 😊

File: pkg/ui/react-app/src/thanos/pages/blocks/BlockDetails.tsx
Comment (prmsrswt) [inline pkg/ui/react-app/src/thanos/pages/blocks/BlockDetails.tsx:57]: Seems like you changed it by mistake. It should be `Labels:`, right?

File: pkg/ui/react-app/src/thanos/pages/blocks/blocks.module.css
Comment (prmsrswt) [inline pkg/ui/react-app/src/thanos/pages/blocks/blocks.module.css:142]: Instead of using custom css, use the `Button` component provided by `reactstrap` to be consistent. You can wrap the `Button` in `a` tag as you need it.

File: pkg/ui/react-app/src/thanos/pages/blocks/helpers.ts
Comment (prmsrswt) [inline pkg/ui/react-app/src/thanos/pages/blocks/helpers.ts:59]: You can do `JSON.stringify(blob, null, 2)` to get properly indented json as output.

File: pkg/ui/react-app/src/thanos/pages/blocks/BlockDetails.tsx
Comment (Oghenebrume50) [reply pkg/ui/react-app/src/thanos/pages/blocks/BlockDetails.tsx]: Ah  yes this was a mistake, Fixed.

File: pkg/ui/react-app/src/thanos/pages/blocks/blocks.module.css
Comment (Oghenebrume50) [reply pkg/ui/react-app/src/thanos/pages/blocks/blocks.module.css]: Okay I have corrected this

File: pkg/ui/react-app/src/thanos/pages/blocks/helpers.ts
Comment (Oghenebrume50) [reply pkg/ui/react-app/src/thanos/pages/blocks/helpers.ts]: Done this and it works fine

**Comment by Oghenebrume50** [comment]
@prmsrswt I need help with the markdown link check? I do not understand why it is failing

**Comment by yeya24** [comment]
@Oghenebrume50 Sorry, please ignore this error first. I opened a pr to fix it in https://github.com/thanos-io/thanos/pull/3282.

**Comment by Oghenebrume50** [comment]
> @Oghenebrume50 Sorry, please ignore this error first. I opened a pr to fix it in #3282.

Hi thank you, there seems to be other test failing now though, and I do not think they are related to the markdown links, I also have zero idea how to fix them.

**Comment by Oghenebrume50** [comment]
Everything works fine now once I merged the new changes from master into the branch,  I also updated the screenshot of how the json file looks when viewed. @prmsrswt @yeya24

File: pkg/ui/react-app/src/thanos/pages/blocks/BlockDetails.tsx
Comment (prmsrswt) [inline pkg/ui/react-app/src/thanos/pages/blocks/BlockDetails.tsx:70]: `Download meta.json` will be more apt here.

File: pkg/ui/react-app/src/thanos/pages/blocks/BlockDetails.tsx
Comment (prmsrswt) [inline pkg/ui/react-app/src/thanos/pages/blocks/BlockDetails.tsx:68]: Add a `<hr />` before this to seperate out this section.

**Review by prmsrswt — APPROVED** [review]
Just some small nits, LGTM after this.

File: CHANGELOG.md
Comment (squat) [inline CHANGELOG.md:13]: Tinnnny nit: add a period at the end of this sentence

**Review by squat — APPROVED** [review]
Lgtm :))

**Review by squat — APPROVED** [review]

File: CHANGELOG.md
Comment (Oghenebrume50) [reply CHANGELOG.md]: Done.

**Pull request merged**
